pkg/webhooks/admission/jobs/validate/admit_job.go

```diff
@@ -194,6 +199,50 @@ func validateJob(job *v1alpha1.Job, reviewResponse *v1beta1.AdmissionResponse) s
 	return msg
 }
 
+func validateJobUpdate(old, new *v1alpha1.Job) error {
+	var totalReplicas int32
+	for _, task := range new.Spec.Tasks {
+		if task.Replicas < 0 {
+			return fmt.Errorf("'replicas' must be >= 0 in task: %s", task.Name)
+		}
+		// count replicas
+		totalReplicas = totalReplicas + task.Replicas
+	}
+	if new.Spec.MinAvailable > totalReplicas {
+		return fmt.Errorf("'minAvailable' must not be greater than total replicas")
+	}
+	if new.Spec.MinAvailable < 0 {
+		return fmt.Errorf("'minAvailable' must be >= 0")
+	}
+
+	if len(old.Spec.Tasks) != len(new.Spec.Tasks) {
+		return fmt.Errorf("job updates may not add or remove tasks")
+	}
+	// other fields under spec are not allowed to mutate
+	new.Spec.MinAvailable = old.Spec.MinAvailable
+	for i := range new.Spec.Tasks {
+		new.Spec.Tasks[i].Replicas = old.Spec.Tasks[i].Replicas
+	}
+
+	// jon controller will update the pvc name if not provided
+	for i := range new.Spec.Volumes {
```

Comment:
that means we did not check volumes update here?

Reply:
Nope, we do not check the volume name here, it is auto generated by job controller.